src/api/plugins/tdbp/warren_tdbp/indicators.py

```diff
@@ -0,0 +1,259 @@
+                action_cohort = actions.iloc[idx]["actor.account.name"]
+
+                if (
+                    0.1 * temp_dynamic_cohort_size <= action_cohort
+                    and action_cohort >= self.dynamic_cohort_size
+                ):
+                    active_actions_iris.append(action_id)
+
+            if len(active_actions_iris) < self.active_actions_min:
+                since -= timedelta(days=1)  # step back from one day
+            else:
+                self.sliding_window = DatetimeRange(since=since, until=self.until)
+                self.dynamic_cohort = temp_dynamic_cohort
+                self.active_actions = self._fetch_active_actions_context(
+                    active_actions_iris, temp_dynamic_cohort_size
+                )
+
+                return SlidingWindow.construct(
+                    **{
+                        "window": self.sliding_window,
+                        "active_actions": self.active_actions,
+                        "dynamic_cohort": self.dynamic_cohort,
+                    }
+                )
+
+        return 0
+
+    def _fetch_active_actions_context(
+        self, statements, iris, dynamic_cohort_size: int
+    ) -> None:
+        """Retrieves contextual information about active actions."""
+        active_actions: List[ActiveAction] = []
+
+        for iri in iris:
+            activation_date = min(
+                statements.loc[(statements["object.id"] == iri)]["timestamp"]
+            )
+            type = statements.loc[(statements["object.id"] == iri)][
+                "context.extensions.http://lrs.learninglocker.net/define/extensions/info.event_name"
+            ].unique()
+
+            cohort = statements.loc[(statements["object.id"] == iri)][
+                "actor.account.name"
+            ].unique()
+
+            score = len(cohort) / dynamic_cohort_size
+
+            active_actions.append(
+                ActiveAction(
+                    id=iri,
+                    activation_date=activation_date,
+                    type=type,
+                    cohort=cohort,
+                    score=score,
+                )
+            )
+
+        return active_actions
+
+
+class ScoreIndicator(SlidingWindowIndicator):
+    """TODO"""
+
+    def __init__(
+        self,
+        course_id: str,
+        student_id: Optional[str] = None,
+        until: Optional[datetime] = None,
+        sliding_window_size: int = 15,
+        active_actions_min: int = 6,
+        dynamic_cohort_size: int = 3,
+    ):
+        """Initialize ScoreIndicator."""
+        super().__init__(
+            course_id,
+            until,
+            sliding_window_size,
+            active_actions_min,
+            dynamic_cohort_size,
+        )
+        self.student_id = student_id
+
+    def fetch(self) -> List[List[StudentScore]]:
+        """TODO"""
+        # Compute active actions indicator
+        self.compute()
```

Comment:
```suggestion
        sliding_window = SlidingWindowIndicator(self.course_id, ...)
        sliding_window.compute()
```

Reply:
The `sliding_window` instantiation and calculation can also be isolated in a private method, or better: a mixin that will be used in other indicators.